contracts/ProjectValidation.sol

```diff
@@ -0,0 +1,180 @@
+    }
+
+    struct Checker {
+        address addr;
+        bool signature;
+        bool signed;
+        bool presence;
+    }
+
+    address public starter;
+
+    address public manager;
+
+    Checker public checker;
+
+    address public exchangerContract;
+
+    bool public projectClosed;
+
+    bool public projectCompleted;
+
+    uint public additionalTokenRate;
+
+    uint public projectBalance;
+
+    uint public workerRatio = 2;
+
+    uint public amountForRoots = 0;
+
+    SimpleToken public fundTokens;
+
+    address[] public workers;
+
+    mapping (address => uint) public workersBalances;
+
+    mapping (address => bool) public signatures;
+
+    event Singed(address who);
+    event Closed();
+    event StateChanged(Stages previous, Stages current);
+
+    Stages public stage = Stages.collectingSignatures;
+
+    function ProjectValidation(
+        address _manager,
+        address _checker,
+        address _exchangerContract,
+        address[] _workers,
+        address fundTokenAddress
+    ) {
+        starter = msg.sender;
+        manager = _manager;
+        checker = Checker(
+            _checker,
+            false,
+            false,
+            false
+        );
+        exchangerContract = _exchangerContract;
+        fundTokens = SimpleToken(fundTokenAddress);
+        workers = _workers;
+        signatures[starter] = false;
+        signatures[manager] = false;
+        for (uint8 i = 0; i < workers.length; i++) {
+            workersBalances[workers[i]] = 0;
+        }
+    }
+
+    modifier onlyValidator() {
+        require(msg.sender == manager || msg.sender == starter || msg.sender == checker.addr);
+        _;
+    }
+
+    modifier onlyChecker(){
+        require(msg.sender == checker.addr && checker.presence);
+        _;
+    }
+
+    modifier atStage(Stages _stage) {
+        require(stage == _stage);
+        _;
+    }
+
+    function changeStateTo (Stages _stage) internal {
+        StateChanged(stage, _stage);
+        stage = _stage;
+    }
+
+    modifier afterExecutingGoToState(Stages _stage){
+        _;
+        changeStateTo(_stage);
+    }
+
+    function sign() onlyValidator atStage(Stages.collectingSignatures) returns (bool signed) {
+        signatures[msg.sender] = true;
+        return true;
+    }
+
+    function checkerSign(bool signature) onlyChecker atStage(Stages.checkerWork) afterExecutingGoToState(Stages.readyToComplete) {
+        checker.signed = true;
+        checker.signature = signature;
+    }
+
+    function checkerCall() internal {
+        changeStateTo(Stages.checkerWork);
+        checker.presence = true;
+    }
+
+    function getWorkerBalance(address worker) returns (uint balance) {
+        uint fundWorkerBalance = workersBalances[worker];
+        balance = fundWorkerBalance * 2 / 3 * additionalTokenRate;
+    }
+
+    function tryToCompleteProject() onlyValidator {
+        if ( ( !signatures[starter] || !signatures[manager] ) && !checker.presence) {
```

Comment:
In my opinion, equivalent expression looks nicer and clearer:
` ! ( (signatures[starter] && signatures[manager] ) || checker.presence)`
What do you think?
Could you please describe what case you want to handle by this expression?

Reply:
Yes, sure, I think that looks simpler. When we call tryToCompleteProject() we check case about when there is disagreement between validators(starter && manager), for example, manager sign the project, but starter no. In this case, we cant complete the project and we will call checker.